django/db/models/fields/__init__.py

```diff
@@ -828,18 +828,17 @@ def get_flatchoices(self, include_blank=True,
         first_choice = blank_choice if include_blank else []
         return first_choice + list(self.flatchoices)
 
+    @warn_about_renamed_method('Field', '_get_val_from_obj', 'value_from_object',
+                               RemovedInDjango21Warning)
     def _get_val_from_obj(self, obj):
-        if obj is not None:
```

Comment:
I understand your reservations and I apologise for my previous outburst, but the test is nonsensical. I can understand `getdefault`-like behaviour in a lot of situations, but getting the default value for a field from `None` isn't something that I think would ever be warranted in code.

Now being concerned about backwards compatibility is justified (and I'm completely for it), but the level of backwards compatibility you're asking me to preserve here would effectively make _any_ code change _anywhere in the library_ impossible. Every bug that exists in the project would have to be preserved because there _might exist_ code that depends on said buggy behaviour.

Now I understand the old adage regarding pull requests -- "change a line, argue for a week, change a million lines and it'll be accepted without question". I knew that going into this and I also knew that this was the first PR that I was going to see through to merge, so I was happy to argue more than I usually would.